systematics/genptshape/ComputePtGenShapeWeights.py

```diff
@@ -7,90 +7,115 @@
+        hPtGenD = contPtShape.Project(0, 0)
+        hPtGenD.SetName('hPtGenD')
+    else:
+        print(f'ERROR: D specie {Bspecie} not implemented for CF outputs! Exit')
 hPtGenD.SetDirectory(0)
 hPtGenD.Sumw2()
-hPtGenD.Rebin(args.rebin)
+hPtGenD.Rebin(rebin)
 hPtGenD.Scale(1./hPtGenD.Integral())
 
-if args.Bspecie:
-    if args.Bspecie not in ['Bs', 'Bplus', 'Bzero', 'Lb']:
-        print(f'ERROR: B specie {args.Bspecie} not supported! Only Bs, Bplus, Bzero, Lb are supported! Exit')
-    if args.Bspecie == 'Bs':
-        hYPtGenB = listGenPtShape.FindObject('hyptBsAllDecay')
-    elif args.Bspecie == 'Bplus':
-        hYPtGenB = listGenPtShape.FindObject('hyptBplusAllDecay')
-    elif args.Bspecie == 'Bzero':
-        hYPtGenB = listGenPtShape.FindObject('hyptB0AllDecay')
-    elif args.Bspecie == 'Lb':
-        hYPtGenB = listGenPtShape.FindObject('hyptLbAllDecay')
-    hYPtGenB.SetDirectory(0)
-    hPtGenB = hYPtGenB.ProjectionX('hPtGenB')
+if Bspecie:
+    if Bspecie not in ['Bs', 'Bplus', 'Bzero', 'Lb']:
+        print(f'ERROR: B specie {Bspecie} not supported! Only Bs, Bplus, Bzero, Lb are supported! Exit')
+    if listGenPtShape:
+        if Bspecie == 'Bs':
+            hYPtGenB = listGenPtShape.FindObject('hyptBsAllDecay')
+        elif Bspecie == 'Bplus':
+            hYPtGenB = listGenPtShape.FindObject('hyptBplusAllDecay')
+        elif Bspecie == 'Bzero':
+            hYPtGenB = listGenPtShape.FindObject('hyptB0AllDecay')
+        elif Bspecie == 'Lb':
+            hYPtGenB = listGenPtShape.FindObject('hyptLbAllDecay')
+        hYPtGenB.SetDirectory(0)
+        hPtGenB = hYPtGenB.ProjectionX('hPtGenB')
+        hPtGenB.SetName('hPtGenB')
+    elif suffixCF != '':
+        print(f'ERROR: B specie {Bspecie} not implemented for CF outputs! Exit')
     hPtGenB.SetDirectory(0)
     hPtGenB.Sumw2()
-    hPtGenB.Rebin(args.rebin)
+    hPtGenB.Rebin(rebin)
     hPtGenB.Scale(1./hPtGenB.Integral())
 
 infileGenPtShape.Close()
 
 # default models
-# TODO: consider to make them arguments
-sFONLLD, _ = ReadFONLL('../../models/fonll/FONLL_D_pp5_y05.txt', True)
-if args.Bspecie:
-    sFONLLB, _ = ReadFONLL('../../models/fonll/FONLL_B_pp5_y05.txt', True)
-
-if args.PbPb:
-    if args.Dspecie == 'Ds':
-        sTAMU, _ = ReadTAMU('../../models/tamu/Ds_TAMU_RAA_5TeV_010.txt')
-        sPHSD, _ = ReadPHSD('../../models/phsd/Ds_PHSD_RAA_5TeV_010.txt')
-        sCatania, _ = ReadCatania('../../models/catania/Ds_Catania_RAA_5TeV_010.dat')
-    else:
-        sTAMU, _ = ReadTAMU('../../models/tamu/D_TAMU_RAA_5TeV_010.txt')
-        sPHSD, _ = ReadPHSD('../../models/phsd/D_PHSD_RAA_5TeV_010.txt')
-        sCatania, _ = ReadCatania('../../models/catania/D_Catania_RAA_5TeV_010.dat')
-    sGossiaux, _ = ReadMCatsHQ('../../models/mcatshq/D_Gossiaux_RAA_5TeV_010.txt')
+isPbPb = False
+sFONLLD, _ = ReadFONLL(shapesD['fonll']['file'], True)
+if Bspecie:
+    sFONLLB, _ = ReadFONLL(shapesB['fonll']['file'], True)
+for shape in shapesD:
```

Comment:
This loop looks unnecessary if you keep the different if statements

Reply:
Done